kernel: limit hardware crypto accelerator modules to proper targets

abajk opened the pull request:
CONFIG_CRYPTO_DEV_PADLOCK depends on X86. This driver only makes sense on X86.
CONFIG_CRYPTO_DEV_GEODE depends on X86_32. This driver only makes sense on X86\geode.
CONFIG_CRYPTO_DEV_TALITOS depends on FSL_SOC. This driver only makes sense on Freescale(NXP) SoCs.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

hauke commented:
I applied 2 of these 3 patches. After I applied 446c861612f9a3a8969d70ab1f8bc7a0c1d1118b make oldocnfig shows these errors:
```
hauke@hauke-t480:~/openwrt/openwrt$ make oldconfig 
tmp/.config-package.in:25058:error: recursive dependency detected!
tmp/.config-package.in:25058:	choice <choice> contains symbol PACKAGE_libssh2
tmp/.config-package.in:25039:	symbol PACKAGE_libssh2 depends on LIBSSH2_MBEDTLS
tmp/.config-package.in:25065:	symbol LIBSSH2_MBEDTLS is part of choice <choice>
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

tmp/.config-package.in:23638:error: recursive dependency detected!
tmp/.config-package.in:23638:	symbol PACKAGE_libcurl depends on LIBCURL_LDAP
feeds/packages/net/curl/Config.in:57:	symbol LIBCURL_LDAP depends on PACKAGE_libcurl
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

package/libs/openssl/Config.in:278:error: recursive dependency detected!
package/libs/openssl/Config.in:278:	symbol OPENSSL_ENGINE_BUILTIN_PADLOCK depends on PACKAGE_libopenssl
tmp/.config-package.in:23008:	symbol PACKAGE_libopenssl depends on OPENSSL_ENGINE_BUILTIN_PADLOCK
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

tmp/.config-package.in:25606:error: recursive dependency detected!
tmp/.config-package.in:25606:	symbol PACKAGE_luci-ssl-openssl depends on PACKAGE_libustream-openssl
tmp/.config-package.in:25265:	symbol PACKAGE_libustream-openssl is selected by PACKAGE_luci-ssl-openssl
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

tmp/.config-package.in:38537:error: recursive dependency detected!
tmp/.config-package.in:38537:	symbol PACKAGE_collectd-mod-mqtt depends on PACKAGE_collectd-mod-mqtt
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

#
# No change to .config
#
```

CodeFetch commented:
The issue is actually that libopenssl depends on HW crypto. I think this should be resolved there instead of depending on a stub...
https://github.com/openwrt/openwrt/blob/ec4852f7313cc377785ce27cac51a71890ab20d6/package/libs/openssl/Makefile#L98

abajk committed:
kernel: limit crypto-hw-padlock to the x86

CONFIG_CRYPTO_DEV_PADLOCK depends on X86. This driver only makes sense on X86.

Signed-off-by: Aleksander Jan Bajkowski <A.Bajkowski@stud.elka.pw.edu.pl>

CodeFetch commented:
@abajk Why did you close it?

abajk commented:
@CodeFetch Currently, I have no idea how to fix this circular dependency.
